### PR TITLE
ignore error on logout and destroy the session

### PR DESCRIPTION
**Pull request by jh-bate**
this relates an error seen when a session times out, then the user clicks "log out", the app hangs. It should re-direct to the login page.

@jebeck if you could have a quick look?

### CODE REVIEW

File: app/core/api.js [hunk 1]
```diff
@@ -112,14 +112,17 @@ api.user.logout = function(cb) {
   api.log('POST /user/logout');
 
   if (!api.user.isAuthenticated()) {
+    api.log('not authenticated but still destroySession');
+    tidepool.destroySession();
     return;
```
Comment (kentquirk) [inline app/core/api.js]: You still need to call cb here -- you can't just return.

**Comment by jebeck** [comment]
Other than @kentquirk's comment, looks good to me.

**Comment by jebeck** [comment]
Oh but also plz update from master @jh-bate

File: app/core/api.js (hunk 1 above)
Comment (jh-bate) [inline app/core/api.js]: hmm i don't think it was even there before? just adding

**Pull request merged**